modules/assignment/src/main/java/com/intuit/wasabi/assignment/impl/AssignmentsImpl.java

```diff
@@ -607,7 +609,7 @@ protected Assignment getAssignment(User.ID userID, Application.Name applicationN
         Assignment assignment = getAssignment(experimentID, userID, context, userAssignments, bucketList);
         if (assignment == null || assignment.isBucketEmpty()) {
             if (createAssignment) {
-                if (experiment.getState() == Experiment.State.PAUSED) {
+                if (experiment.getState() == Experiment.State.PAUSED || !checkUserCapForRapidExperiment(experiment)) {
```

Comment:
Regarding, Place where we are checking the user assignment cap: it would be great if experiment caps are retrieved from database asynchronously and then checked here synchronously in a loop one by one in a priority order...